compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -527,16 +518,12 @@ impl<'f> LoopInvariantContext<'f> {
                     .any(|predecessor| {
                         self.post_dom_frontiers.is_control_dependent(*predecessor, nested.header)
                     });
+
             if nested_loop_is_control_dep {
-                loop_context.current_block_control_dependent = true;
-                // Mark all blocks in the nested loop as control dependent to avoid redundant checks
-                // for each of these blocks when they are later visited during hoisting.
-                // This is valid because control dependence of the loop header implies dependence
-                // for the entire loop body.
-                loop_context.nested_loop_control_dependent_blocks.extend(nested.blocks.iter());
```

Comment:
I feel it would be nicer to still accept a loop_context and extend the `nested_loop_control_dependent_blocks` set here. We can then just return a `ControlDependenceResult`, remove the `InNestedLoop(&'a Loop)` variant, and remove the lifetime. 

It seems cleaner that `find_control_dependent_nested_loop` then returns `ControlDependenceResult::InSomeNestedLoop` or `ControlDependenceResult::No`. The result of a control dependent nested loop then matches this check:
```
        if loop_context.nested_loop_control_dependent_blocks.contains(&block) {
            return ControlDependenceResult::InSomeNestedLoop;
        }
```

Reply:
Tried to find a middle ground by passing `&mut loop_context.nested_loop_control_dependent_blocks` to `is_control_dependent_post_pre_header` and got rid of `ControlDependencyResult`. This way it's clear we are passing a simple cache, not using any logic.